server/routerlicious/packages/services/src/mongodb.ts

```diff
@@ -135,10 +206,26 @@ export class MongoCollection<T> implements core.ICollection<T> {
 
         await this.collection.updateMany(filter, update, options);
     }
+
+    private async requestWithRetry<TOut>(request: () => Promise<TOut>, callerName: string): Promise<TOut> {
+        return requestWithRetry<TOut>(
```

Comment:
My understanding was that `requestWithRetry` is intentionally an alternative to `runWithRetry` designed for HTTP requests. Why use `requestWithRetry` rather than `runWithRetry` for a non-HTTP request?

Reply:
I don't think `requestWithRetry` was for http call only:
![image](https://user-images.githubusercontent.com/3719246/200659568-4aa5482e-acbf-4fdf-98b4-dc60ccf862f5.png)

It used retry on network error as the default should retry logic, but here I used the analyze results based on error.

It specified for `requestWithRetry`:
> The difference between {@link requestWithRetry} and {@link runWithRetry} is that {@link runWithRetry} allows the user to decide whether to ignore the error or not, on a "fire and forget" fashion. That makes the return type of {@link runWithRetry} be slightly different, as `undefined` is a possible return value. That is not the case for {@link requestWithRetry}, which focuses on requests/operations where the user always wants the error to be bubbled up.

I don't want `undefined` and I don't want the "fire and forgot"/ignore error for this case  so used `requestWithRetry`

Reply:
Makes sense! Thanks for explaining!